api/sendMail.ts

```diff
@@ -0,0 +1,53 @@
+import axios, { AxiosError } from "axios";
+import {
+  IUserMailErrorResponse,
+  IUserMailSuccessfullResponse,
+  IUserMailRequest,
+  IUserMailResponse,
+  IUserMailSend,
+  IUserRatingRequest,
+  IUserRatingResponse
+} from "../types/Mail";
+import { Endpoints } from "../types/Endpoints.enum";
+
+const getErrorObject = (errorMessage: string): IUserMailErrorResponse => ({
+  errorMessage,
+});
+
+const isErrorResponse = (
+  data: unknown
+): data is IUserMailErrorResponse => {
+  return (data as IUserMailErrorResponse).errorMessage !== undefined;
+};
+
+export class SendMail implements IUserMailSend {
+  async send(request: IUserMailRequest): Promise<IUserMailResponse> {
+    try {
+      const resp = await axios.post(Endpoints.SEND, request);
+      const data = resp.data as IUserMailSuccessfullResponse;
+      return data;
```

Comment:
You can return that data in the same time.
`return resp.data as UserMailSuccessfullResponse;`